src/future.ts

```diff
@@ -0,0 +1,209 @@
+import { assert } from './util'
+import Context from './context'
+import { Middleware } from './middleware'
+
+type ReplyContext = { [key in keyof Context & `reply${string}`]: Context[key] }
+
+function makeReply<
+  C extends Context,
+  E extends { reply_to_message_id?: number }
+>(ctx: C, extra?: E) {
+  const ret = extra || ({} as E)
+  ret.reply_to_message_id = ctx.message?.message_id
+  return ret
```

Comment:
Avoid mutating users' arguments:
```suggestion
  const reply_to_message_id = ctx.message?.message_id
  return { reply_to_message_id, ... extra }
```